### PR TITLE
Cleaner way to have highlighted text on invalid address

### LINKED ISSUE #20833: Hardcoded color values on theme core switchEditAddressButtonColor
#### Describe the bug

https://github.com/PrestaShop/PrestaShop/blob/f3ea0384869e849e58a5663e4c8446733f27a0b3/themes/_core/js/checkout-address.js#L94-L103

#### Expected behavior
These values must be defined in theme. For example on data-color attribute or adding class.

#### Steps to Reproduce

0. Enable custom theme with diferent colors
1. Create invalid address
2. Select invalid address on cart (step 2)
3. Then, edit button change to hardcoded value.

**Screenshots**
Before change addres to invalid one
image

After
image

#### Additional information

* PrestaShop version: 1.7.6 and probably up to develop
* PHP version: 7.1

### PR DESCRIPTION
**Pull request by kpodemski**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Details #20833 
| Type?         | improvement
| Category?     | FO
| BC breaks?    | visually - no, code? yes, but this shouldn't be an issue tbh
| Deprecations? | no
| Fixed ticket? | Fixes #20833 
| How to test?  | 1. add new address, make sure to have at least 2 or 3 addresses<br>2. go to your db, `PREFIX_address`, delete firstname, lastname or any other required field from one of addresses you have<br>3. go to checkout, you should see "Address is incomplete" message after selecting invalid address<br>4. before and after my change "Edit" link should be highlighted with blue color
| Sponsor company  | impSolutions

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/20909)

### CODE REVIEW

**Review by Progi1984 — APPROVED** [review]

File: themes/_core/js/checkout-address.js [hunk 1]
```diff
@@ -92,14 +92,14 @@ $(window).on('load', () => {
...
-  let color = '#7a7a7a';
+  const addressBtn = $('#id-address-' + type + '-address-' + id + ' a.edit-address');
+  const classesToToggle = ['text-info', ' address-item-invalid'];
 
   if (enabled) {
-    $('#' + type + '-addresses a.edit-address').prop('style', 'color: #7a7a7a !important');
-    color = '#2fb5d2';
+    addressBtn.addClass(classesToToggle);
+  } else {
+    addressBtn.removeClass(classesToToggle);
```
Comment (PululuK) [inline themes/_core/js/checkout-address.js]: **Suggestion** : Use `toggleClass` see https://api.jquery.com/toggleclass/. Something like :
```js
addressBtn.toggleClass(classesToToggle, enabled);
```

**Review by PierreRambaud — CHANGES_REQUESTED** [review]
Agree with @PululuK (I was going to do the same comment :sweat_smile: )

**Review by Progi1984 — CHANGES_REQUESTED** [review]
Good optimization. Thanks @PululuK

File: themes/_core/js/checkout-address.js (hunk 1 above)
Comment (kpodemski) [reply themes/_core/js/checkout-address.js]: @PululuK thank you, i didn't know about second param :D new-school jquery :D

**Comment by kpodemski** [comment]
@Progi1984 done

**Review by PierreRambaud — DISMISSED** [review]

**Review by PululuK — DISMISSED** [review]

**Review by Progi1984 — DISMISSED** [review]

**Comment by khouloudbelguith** [comment]
Hi @kpodemski,

I found this issue following these steps:
1. You have already 3 incomplete address
2. Go to the checkout page
3. In the **Addresses** step => 
4. "Your Invoice Address" => select the third address
5. "Shipping Address" => select the second address
6. see the error, Edit for the first address, it is highlighted with blue color
image
7. "Your Invoice Address" => select second address
8. see the error, Edit for the third address, it is highlighted with blue color
image

I attached a screen record
https://drive.google.com/file/d/1AArNyW-dBGuph_kpAEk4f1doOA8IaY1q/view?usp=sharing

In the develop branch => ok.

Thanks to check and feedback.

**Comment by kpodemski** [comment]
Thank you @khouloudbelguith . Nice catch.

Now code is clearing highlight from each address of given type.

**Comment by kpodemski** [comment]
@Progi1984 @PierreRambaud can you re-run Travis?

**Comment by Progi1984** [comment]
@kpodemski I'll do it.

**Review by Progi1984 — APPROVED** [review]

**Review by PierreRambaud — APPROVED** [review]

**Review by atomiix — APPROVED** [review]

**Comment by khouloudbelguith** [comment]
Hi @kpodemski,

It is ok :heavy_check_mark: 
image

https://drive.google.com/file/d/1TuS0y5_BtUrAx4UdwW7XrZqA0vVbDx0k/view?usp=sharing

Thanks!

**Comment by atomiix** [comment]
Thanks @kpodemski !

**Comment by davidglezz** [comment]
Oops, I hadn't seen this!
I wanted to show my opposition to using the 'text-info' class, since it indicates color, in classic theme accent color is similar to info, but not in other themes. imagine a dark theme.

What do you think if I make a PR with only 'address-item-invalid' class (removing 'text-info') and define the colors in the css?
```css
.address-item-invalid {
     color: # 2fb5d2;
}
```

By the way, very good solution! @kpodemski

**Comment by kpodemski** [comment]
@davidglezz if you specify more detailed selector like a.address-item-invalid your CSS should have priority, which means that authors of dark theme can easily change this color

if you think this will help theme authors go and create PR 👍

**Pull request merged**